### PR TITLE
Sets the date_created_gmt field when creating a page

### PR DESCRIPTION
**Pull request by elibud**
The `date_created_gmt` field is used for the Page List sorting so it should be filled when creating a new page.

**Fixes** #7148 

**To test:**

First make sure you can replicate the issue on develop.
Then checkout this branch and check that you can't :)
I am not 100% sure this is the right solution, but it works wonderfully well and is the only thing I can think of doing because the field `date_created_gmt` is what is being used to sort the Page List.

Needs review: @koke

### CODE REVIEW

**Comment by koke** [comment]
I don't understand what's going on. Adding the date would fix the sorting, but why wasn't it updating the title?

**Comment by elibud** [comment]
Honestly I don't know, I find the way that `PageListViewController` works a bit hard to follow. What is more strange is that it used to not work only for the first page added, if you left the list and went back in after adding at least one new page everything worked fine.

**Comment by elibud** [comment]
@koke  as per your suggestions I have updated the filter predicate (not sorting, the commit message is incorrect). 
Also now, when the editor is closed, we update the filter to the status of the last saved page as we do on posts, so if you try to publish while offline, the draft is saved and we to to the draft list.
